FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_keep_alive/DemoTasks/KeepAliveMQTTExample.c

```diff
@@ -311,6 +333,18 @@ static uint16_t usSubscribePacketIdentifier;
  */
 static uint16_t usUnsubscribePacketIdentifier;
 
+/**
+ * @brief Status of latest Subscribe ACK;
+ * it is updated every time the callback function processes a Subscribe ACK.
+ */
+static MQTTSubAckStatus_t xGlobalSubAckStatus = MQTTSubAckFailure;
+
+/**
+ * @brief Topic to subscribe.
+ * Used to re-subscribe to topics that failed initial subscription attempts.
+ */
+static MQTTSubscribeInfo_t xGlobalSubscribeInfo;

```

Comment:
I am concerned about maintaining state here. This means if I call subscribe again before the first subscribe is completed I will break things? What mechanisms do we have in place to prevent this? How does QOS1 affect this, retransmissions take time where you are not allowed to make another subscribe? In the spec you can have multiple subscribes in flight, this makes that impossible so it is a restrictive constraint.